--- a/r4300/fpu.h
+++ b/r4300/fpu.h
@@ -38,7 +38,7 @@
   {
     static const unsigned int msRound[4] = { _RC_NEAR, _RC_CHOP, _RC_UP, _RC_DOWN };
     unsigned int oldX87, oldSSE2;
-    __control87_2(msRound[RoundType], _MCW_RC, &oldX87, &oldSSE2);
+    _controlfp_s(&oldX87, msRound[RoundType], _MCW_RC);
   }
   static __inline double round(double x) { return floor(x + 0.5); }
   static __inline float roundf(float x) { return (float) floor(x + 0.5); }
