--- a/vbam/gba/GBAcpu.h
+++ b/vbam/gba/GBAcpu.h
@@ -7,7 +7,11 @@
 extern int thumbExecute(GBASystem *);

 #ifdef __GNUC__
-# define INSN_REGPARM __attribute__((regparm(2)))
+# if defined(__i386__) || defined(__x86_64__)
+#  define INSN_REGPARM __attribute__((regparm(2)))
+# else
+#  define INSN_REGPARM /*nothing*/
+# endif
 # define LIKELY(x) __builtin_expect(!!(x),1)
 # define UNLIKELY(x) __builtin_expect(!!(x),0)
 #else
