--- a/src/synth_core.cpp
+++ b/src/synth_core.cpp
@@ -328,7 +328,7 @@

         b4 -= b4*b4*b4 * (1.0f/6.0f); // clipping
         b4 -= fcdcoffset; // un-bias
-        b[4] = b4 - fcdcoffset;
+        b[4] = b4;
         b[0] = realin;

         return b4;
@@ -850,7 +850,7 @@
             output(dest + i, gain * x);
         }

-        flt = nf;
+        nf = flt;
         nseed = seed;
     }

